Use Java 8 map.putIfAbsent and map.computeIfAbsent #8215

pedrolarben opened the pull request:
Part of #8215 <!-- Fixes #8215 -->
_Ready for review_

I have change some methods in order to use the java8 methods: map.putIfAbsent, map.computeIfAbsent and map.computeIfPresent. 

I have changed all the possible methods that I have found, except for those that have already being fixed in the ongoing PR #8705  

<!-- Tell us how you solved the issue. -->
<!-- If there are things you want the reviewers to focus on, include them here as well. -->
<!-- This portion can be skipped if the fix is trivial. -->

wenmogu commented on src/main/java/teammates/common/datatransfer/InstructorPrivileges.java:
```diff
@@ -269,10 +269,10 @@ private void updatePrivilegesInSessionLevel(String sectionName, String sessionNa
             addSectionWithDefaultPrivileges(sectionName);
-            this.sessionLevel.put(sectionName, new LinkedHashMap<String, Map<String, Boolean>>());
-        }
+            return new LinkedHashMap<String, Map<String, Boolean>>();
```
you can simply return `new LinkedHashMap<>`

pedrolarben replied:
I decided to return `new LinkedHashMap<String, Map<String, Boolean>>` insted of simply `new LinkedHashMap` in order to try to not change the old code. In the old code we were putting a `new LinkedhashMap<String, Map<String, Boolean>>` in the map `this.sessionLevel` so I decided to do the same.


**Old Code**
```java
    private void verifyExistenceOfsectionName(String sectionName) {
        if (!this.sessionLevel.containsKey(sectionName)) {
            addSectionWithDefaultPrivileges(sectionName);
            this.sessionLevel.put(sectionName, new LinkedHashMap<String, Map<String, Boolean>>());
        }
    }
```
**New Code**
```java
    private void verifyExistenceOfsectionName(String sectionName) {
        this.sessionLevel.computeIfAbsent(sectionName, key -> {
            addSectionWithDefaultPrivileges(sectionName);
            return new LinkedHashMap<String, Map<String, Boolean>>();
        });
    }
```

wenmogu replied:
Yeah I know, but that part is unnecessary even in the old code. Not all parts of the old code e correct :P

wenmogu replied:
Hi @pedrolarben you r right, they r equivalent. We shall change it to make the code cleaner :)

pedrolarben commented:
Changes made

pedrolarben commented:
@wenmogu I'm having problem executing the npm lint test, should I open a new issue to ask for help?
`Error: Cannot find module 'webpack-merge'`

pedrolarben commented:
@wenmogu  I have changed the code. I have fixed the code style


**Tests**
- [X] Code Style test  
- [X] Local tests 
- [ ] CI tests
 
I need help with the CI tests because there are some tests failing :(

wenmogu commented:
oh? as long as you can pass the Travis CI test online here :)
I know there are some random tests failing e.g. some EditStudentUITest or RankQuestionUITest... try run the failed tests a few times generally it will pass eventually (if you didnt modify any relevant code)... i guess its due to some time lag that selenium couldnt handle at times...

wenmogu commented:
hi @wkurniawan07 can pls take a look at this? shld be a fast read :D 

I think after this PR and the other ongoing PR #8705  this issue can be officially closed. I guess another first timer issue about replacement using `computeIfPresent` can be done as there are quite some places (arnd 10) that can be shortened using this method.

pedrolarben commented:
Change done @wkurniawan07
